plutus-core/generators/Language/PlutusCore/Generators/NEAT/Spec.hs

```diff
@@ -114,12 +123,14 @@ prop_typePreservation tyG tmG = do
 
   -- Check if the converted term, when evaluated by CK, still has the same type:
 
-  tmCK <- withExceptT CkP $ liftEither $ evaluateCk defBuiltinsRuntime tm
+  tmCK <- withExceptT CkP $ liftEither $
+    evaluateCk defBuiltinsRuntime tm `catchError` handleError ty
   withExceptT TypeError $ checkType tcConfig () tmCK (Normalized ty)
 
   -- Check if the converted term, when evaluated by CEK, still has the same type:
-
-  tmCEK <- withExceptT CekP $ liftEither $ evaluateCek defBuiltinsRuntime tm
+  -- NOTE: the CEK machine doesn't respect this so we are just going through the motions here
+  tmCEK <- withExceptT CekP $ liftEither $ evaluateCek
+    defBuiltinsRuntime tm `catchError` handleError ty
```

Comment:
Maybe remove the test then? It doesn't seem we're going to patch up the CEK machine to fix that problem any time soon.